### PR TITLE
[ticket/15719] Add core events to viewtopic to edit the post_list sql query

### PR DESCRIPTION
**Pull request by toxyy**
Event added in viewtopic.php so you can modify post_list sql

The query, sql limit, sql start, sort keys and days are editable

PHPBB3-15719

Checklist:

- [ ] Correct branch: master for new features; 3.2.x for fixes
- [ ] Tests pass
- [x] Code follows coding guidelines: [master](https://area51.phpbb.com/docs/dev/master/development/coding_guidelines.html) and [3.2.x](https://area51.phpbb.com/docs/dev/3.2.x/development/coding_guidelines.html)
- [x] Commit follows commit message [format](https://area51.phpbb.com/docs/dev/3.2.x/development/git.html)

Tracker ticket (set the ticket ID to **your ticket ID**):

https://tracker.phpbb.com/browse/PHPBB3-15719

### CODE REVIEW

**Comment by toxyy** [comment]
I'm the only participant this time, I edited it so that the title and few lines after are less than 79 characters, and i fixed the comment block up a little for the event.  Thanks 3Di!

**Comment by toxyy** [comment]
Why does that one travis check keep failing?

**Comment by 3D-I** [comment]
> FILE: /home/travis/build/phpbb/phpbb/phpBB/**viewtopic.php**
> ----------------------------------------------------------------------
> FOUND 3 ERRORS AFFECTING 3 LINES
> ----------------------------------------------------------------------
>  **1151** | ERROR | [x] **Whitespace found at end of line**
>       |       |     (Squiz.WhiteSpace.SuperfluousWhitespace.EndLine)
>  **1172** | ERROR | [x] **Whitespace found at end of line**
>       |       |     (Squiz.WhiteSpace.SuperfluousWhitespace.EndLine)
>  **1173** | ERROR | [x] **Whitespace found at end of line**
>       |       |     (Squiz.WhiteSpace.SuperfluousWhitespace.EndLine)

File: phpBB/viewtopic.php [hunk 1]
```diff
@@ -1148,6 +1148,29 @@
 		" . (($join_user_sql[$sort_key]) ? 'AND u.user_id = p.poster_id': '') . "
 		$limit_posts_time
 	ORDER BY $sql_sort_order";
+
```
Comment (3D-I) [inline phpBB/viewtopic.php]: Whitespace found at end of line

File: phpBB/viewtopic.php [hunk 2]
```diff
@@ -1148,6 +1148,29 @@
...
+*/
+$vars = array(
+	'sql',
+	'sql_limit',
+	'sql_start',
+	'sort_key',
+	'sort_days',
+	'forum_id',
+);
+extract($phpbb_dispatcher->trigger_event('core.viewtopic_modify_post_list_sql', compact($vars)));
```
Comment (3D-I) [inline phpBB/viewtopic.php]: Whitespace found at end of line

File: phpBB/viewtopic.php [hunk 3]
```diff
@@ -1148,6 +1148,29 @@
...
+$vars = array(
+	'sql',
+	'sql_limit',
+	'sql_start',
+	'sort_key',
+	'sort_days',
+	'forum_id',
+);
+extract($phpbb_dispatcher->trigger_event('core.viewtopic_modify_post_list_sql', compact($vars)));		
+
```
Comment (3D-I) [inline phpBB/viewtopic.php]: Whitespace found at end of line

File: phpBB/viewtopic.php [hunk 4]
```diff
@@ -1148,6 +1148,29 @@
...
+* Event to modify the SQL query that gets post_list
+*
+* @event core.viewtopic_modify_post_list_sql
+* @var	string	sql			The SQL query to generate the post_list
+* @var	int		sql_limit	The number of posts that the query fetches
+* @var	int		sql_start	The index the query starts to fetch from
+* @var	string	sort_key	Key the posts are sorted by
+* @var	string	sort_days	Display posts of previous x days
+* @var	int		forum_id	Forum ID
+* @since 3.2.2
```
Comment (3D-I) [inline phpBB/viewtopic.php]: @since 3.2.4-rc1

**Comment by toxyy** [comment]
Looks like it doesn't like 3.2.4-rc1

**Comment by 3D-I** [comment]
No.  

This time I don't know..

> Inspecting commit message of commit 7159d07d00e99e2c7a6253df96c628c054deb03f
> Unexpected EOF encountered 

Try to close and reopen this PR to let it start again. 
Use the close button here I mean, then open.

**Comment by 3D-I** [comment]
Same as before, leave it as it is.
Somebody will chime on this and you will get a notification,

**Comment by toxyy** [comment]
That's really weird...

**Comment by kasimi** [comment]
* One of your commit messages doesn't start with `[ticket/15719]`
* `@since` should be `3.2.4-RC1` (upper-case `RC`)

**Comment by toxyy** [comment]
I fixed both of those issues in the last couple of commits.  Changing it to RC1 made the rest pass, but the first still fails.

**Comment by toxyy** [comment]
I did that one through github. But it doesn't really explain why the first travis test is failing with eof?

**Comment by 3D-I** [comment]
As per @kasimi ..
The commit message must contain [ticket/xxx] and PHPBB3-xxx because links to the tracker, each commit.

I think is better to close and make a new PR if you don't know how to amend this one, which it's a bit hard at the beginning, we all once were green.

**Pull request closed without merging**